plugins/inputs/nomad/nomad.go

```diff
@@ -0,0 +1,205 @@
+package nomad
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/config"
+	"github.com/influxdata/telegraf/plugins/common/tls"
+	"github.com/influxdata/telegraf/plugins/inputs"
+)
+
+// Nomad configuration object
+type Nomad struct {
+	URL string `toml:"url"`
+
+	AuthToken       string `toml:"auth_token"`
+	AuthTokenString string `toml:"auth_token_string"`
+
+	ResponseTimeout config.Duration `toml:"response_timeout"`
+
+	tls.ClientConfig
+
+	roundTripper http.RoundTripper
+}
+
+const timeLayout = "2006-01-02 15:04:05 -0700 MST"
+
+var sampleConfig = `
+  ## URL for the Nomad agent
+  # url = "http://127.0.0.1:4646"
+
+  ## Use auth token for authorization. 
+  ## Only one of the options can be set. Leave empty to not use any token.
+  # auth_token = "/path/to/auth/token"
+  ## OR
+  # auth_token_string = "a1234567-40c7-9048-7bae-378687048181"
+
+  ## Set response_timeout (default 5 seconds)
+  # response_timeout = "5s"
+
+  ## Optional TLS Config
+  # tls_ca = /path/to/cafile
+  # tls_cert = /path/to/certfile
+  # tls_key = /path/to/keyfile
+`
+
+func init() {
+	inputs.Add("nomad", func() telegraf.Input {
+		return &Nomad{}
+	})
+}
+
+// SampleConfig returns a sample config
+func (n *Nomad) SampleConfig() string {
+	return sampleConfig
+}
+
+// Description returns a description of the plugin
+func (n *Nomad) Description() string {
+	return "Read metrics from the Nomad API"
+}
+
+func (n *Nomad) Init() error {
+	if n.URL == "" {
+		n.URL = "http://127.0.0.1:4646"
+	}
+
+	if n.AuthToken != "" && n.AuthTokenString != "" {
+		return fmt.Errorf("config error: both auth_token and auth_token_string are set")
+	}
+
+	if n.AuthToken != "" {
+		token, err := os.ReadFile(n.AuthToken)
+		if err != nil {
+			return fmt.Errorf("reading file failed: %v", err)
+		}
+		n.AuthTokenString = strings.TrimSpace(string(token))
+	}
+
+	tlsCfg, err := n.ClientConfig.TLSConfig()
+	if err != nil {
+		return fmt.Errorf("setting up TLS configuration failed: %v", err)
+	}
+
+	if n.ResponseTimeout < config.Duration(time.Second) {
+		n.ResponseTimeout = config.Duration(time.Second * 5)
+	}
```

Comment:
This will not allow the user to set a duration smaller than one second. It would be better to initialize the default in `init` like
```golang
func init() {
	inputs.Add("nomad", func() telegraf.Input {
		return &Nomad{
			ResponseTimeout: 5*time.Second,
		}
	})
}
```
and remove the code here. This allows zero values and thus the user can _force_ "no-timeout" by setting this to zero.

Reply:
Sure, it makes sense.